File: 02_EstructurasControl/Repetitivas/While/Python/Ejemplo02.py
# Procesamiento de un cajero automático: Mantiene un bucle que permite al
# usuario realizar múltiples transacciones hasta que decida salir.
continuar = True
print("Bienvenido al cajero automático.")

while continuar:
    print("Seleccione una opción:")
    print("1. Consultar saldo")
    print("2. Retirar dinero")
    print("3. Depositar dinero")
    opcion = input()

    if opcion == "1":
        print("Su saldo es de $1000.")
    elif opcion == "2":
        print("Ingrese la cantidad a retirar:")
        retiro = input()
        print(f"Ha retirado ${retiro}.")
    elif opcion == "3":
        print("Ingrese la cantidad a depositar:")
        deposito = input()
        print(f"Ha depositado ${deposito}.")
    else:
        print("Opción no válida.")

    print("¿Desea realizar otra transacción? (s/n)")
    respuesta = input().lower()  # Convierte a minúsculas para comparación insensible
    if respuesta != "s":
        continuar = False
        print("Gracias por usar el cajero automático. ¡Hasta luego!")

    input("Presione Enter para continuar...")  # Pausa equivalente a ReadKey()

File: 02_EstructurasControl/Repetitivas/While/C#/Ejemplo-02/Ejemplo02.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _2.While
{
    internal class Program
    {
        static void Main(string[] args)
        {
            //Procesamiento de un cajero automático: Mantiene un bucle que permite al
            //usuario realizar múltiples transacciones hasta que decida salir.
            bool continuar = true;
            Console.WriteLine("Bienvenido al cajero automático.");
            while (continuar)
            {
                Console.WriteLine("Seleccione una opción:");
                Console.WriteLine("1. Consultar saldo");
                Console.WriteLine("2. Retirar dinero");
                Console.WriteLine("3. Depositar dinero");
                string opcion = Console.ReadLine();
                switch (opcion)
                {
                    case "1":
                        Console.WriteLine("Su saldo es de $1000.");
                        break;
                    case "2":
                        Console.WriteLine("Ingrese la cantidad a retirar:");
                        string retiro = Console.ReadLine();
                        Console.WriteLine($"Ha retirado ${retiro}.");
                        break;
                    case "3":
                        Console.WriteLine("Ingrese la cantidad a depositar:");
                        string deposito = Console.ReadLine();
                        Console.WriteLine($"Ha depositado ${deposito}.");
                        break;
                    default:
                        Console.WriteLine("Opción no válida.");
                        break;
                }
                Console.WriteLine("¿Desea realizar otra transacción? (s/n)");
                string respuesta = Console.ReadLine();
                if (respuesta.ToLower() != "s")
                {
                    continuar = false;
                    Console.WriteLine("Gracias por usar el cajero automático. ¡Hasta luego!");
                }

                Console.ReadKey();

            }
        }
    }
}
